revenue_app/views.py

```diff
@@ -146,39 +154,101 @@
 ]
 
 
-class Dashboard(TemplateView):
+class QueriesRequiredMixin():
```

Comment:
What would be an alternative to use a Mixin here? Mixins are fine in _some_ situations but they usually complicate things unnecessarily.

Reply:
Maybe a decorator, but they seem harder to implement. This is based on `LoginRequiredMixin`, which also has a `@login_required` decorator

Reply:
Yes, I thought about a decorator. Maybe we could make it parametrizable so we have a `@require_queries` decorator and you'd use it like `@require_queries(['transactions', 'corrections', 'organizer_sales'])`? It's not _too_ soon to be generic here, imho :).

Reply:
:(